### DISCUSSION TITLE
Command Line Interface (cli) --info option not working as per documentation, always gives invalid choice

### DISCUSSION THREAD

**Discussion by AjayaVeeramalli** [JanssenProject/jans#6572] [Q&A] [answered]
Installed Janssen community edition on Ubuntu 22.04 as per instructions at: https://docs.jans.io/v1.0.20/admin/install/vm-install/ubuntu/

Tried configuring the server using the CLI, tried to CLI' s --info option to start with but that doesn't work as expected. The following is the output: 
**python3 /opt/jans/jans-cli/cli/config_cli.py --info DefaultAuthenticationMethod** 
_usage: config_cli.py [-h] [--host HOST] [--client-id CLIENT_ID] [--client-secret CLIENT_SECRET] [--access-token ACCESS_TOKEN] [--plugins PLUGINS] [-debug] [--debug-log-file DEBUG_LOG_FILE] [--operation-id OPERATION_ID] [--url-suffix URL_SUFFIX] [--info {}] [--op-mode {get,post,put,patch,delete}] [--endpoint-args ENDPOINT_ARGS] [--schema SCHEMA] [-CC CONFIG_API_MTLS_CLIENT_CERT] [-CK CONFIG_API_MTLS_CLIENT_KEY] [--key-password KEY_PASSWORD] [-noverify] [-use-test-client] [--patch-add PATCH_ADD] [--patch-replace PATCH_REPLACE] [--patch-remove PATCH_REMOVE] [-no-color] [--log-dir LOG_DIR] [-revoke-session] [-scim] [-auth] [--data DATA] [--output-access-token] config_cli.py: error: argument --info: invalid choice: 'DefaultAuthenticationMethod' (choose from )_

As per the documentation at: https://docs.jans.io/v1.0.20/admin/config-guide/config-tools/jans-cli/, this should give: 
**Operation ID: get-acrs Description: Gets default authentication method. Operation ID: put-acrs Description: Updates default authentication method. Schema: /components/schemas/AuthenticationMethod

To get sample shema type /opt/jans/jans-cli/config-cli.py --schema <schma>, for example /opt/jans/jans-cli/config-cli.py --schema /components/schemas/AuthenticationMethod**

**Comment by nynymike** [comment]
@devrimyatar Can you check?

**Comment by ossdhaval** [comment]
Hi @AjayaVeeramalli I have the same build and the same command I tried. Seems to be working. To be sure, you can compare the build number that you are using with what is available in the screenshot below :point_down: 

![image](https://github.com/JanssenProject/jans/assets/343411/5e6ae07f-d893-40c4-a30f-285260944d3c)

**Comment by AjayaVeeramalli** [comment]
Thanks @ossdhaval for the response. 

Versions seem to be correct:
administrator@ubuntu-2204:~$ sudo /opt/jans/bin/show_version.py 
Title: 
Version: 
Builddate: 
Build: 

Title: Jans authentication server
Version: 1.0.20
Builddate: 
Build: cc9d64f830ac3a07c7dbcbaafe920386e6fdcb7f

Title: SCIM API Server
Version: 1.0.20
Builddate: 
Build: cc9d64f830ac3a07c7dbcbaafe920386e6fdcb7f

Title: 
Version: 
Builddate: 
Build: 

Title: casa
Version: 1.0.20
Builddate: 
Build: cc9d64f830ac3a07c7dbcbaafe920386e6fdcb7f

Title: Fido2 API Server
Version: 1.0.20
Builddate: 
Build: cc9d64f830ac3a07c7dbcbaafe920386e6fdcb7f

Title: jans-config-api-server
Version: 1.0.20
Builddate: 
Build: 690fa33d511ce5c5ca31ba838ce30ad26c84652b

Is there any other installation / configuration step that I am missing.  By the way, this is what I get when I tried the --schema option:
administrator@ubuntu-2204:~$ /opt/jans/jans-cli/cli/config_cli.py --schema AuthenticationMethod
Traceback (most recent call last):
  File "/opt/jans/jans-cli/cli/config_cli.py", line 1555, in <module>
    main()
  File "/opt/jans/jans-cli/cli/config_cli.py", line 1525, in main
    cli_object = JCA_CLI(host, client_id, client_secret, access_token, test_client, wrapped=False)
  File "/opt/jans/jans-cli/cli/config_cli.py", line 280, in __init__
    self.host += '/jans-config-api'
TypeError: unsupported operand type(s) for +=: 'NoneType' and 'str'

I strongly feel that I might have missed some step. Checked the documentation, but not able to figure out.  Also, just to bring to your attention, I am using Ubuntu 22.04.

**Reply by ossdhaval** [reply]
@AjayaVeeramalli 

- You don't need any other config step to run this successfully. Just that Janssen Server installation should have completed property.
- Ubuntu 22.04 is well supported. Shouldn't be an issue.

May I ask what backend are you using :grey_question:  I have checked this on a system running LDAP backend.

**Comment by AjayaVeeramalli** [comment]
I am using the embedded ldap that comes with janssen installation.

**Reply by ossdhaval** [reply]
@AjayaVeeramalli I see. Not sure why is it giving issues. If it is possible for you, try with latest build and see if results are better. Alternatively, Janssen offers TUI (text-based user interface) to manage configuration. Can you check if you are able to launch TUI? Use `/opt/jans/jans-cli/jans_cli_tui.py` command to launch.

**Comment by AjayaVeeramalli** [comment, marked as answer]
I did the reinstall and tried. The reinstall seems to have fixed some other issues that I was having. However, I was still getting this isssue. But then it just went away when I did the same with sudo. 

The following is the latest: (with and without sudo)
$ python3 /opt/jans/jans-cli/config-cli.py --info DefaultAuthenticationMethod
usage: config-cli.py [-h] [--host HOST] [--client-id CLIENT_ID] [--client-secret CLIENT_SECRET] [--access-token ACCESS_TOKEN]
                     [--plugins PLUGINS] [-debug] [--debug-log-file DEBUG_LOG_FILE] [--operation-id OPERATION_ID]
                     [--url-suffix URL_SUFFIX] [--info {}] [--op-mode {get,post,put,patch,delete}] [--endpoint-args ENDPOINT_ARGS]
                     [--schema SCHEMA] [-CC CONFIG_API_MTLS_CLIENT_CERT] [-CK CONFIG_API_MTLS_CLIENT_KEY] [--key-password KEY_PASSWORD]
                     [-noverify] [-use-test-client] [--patch-add PATCH_ADD] [--patch-replace PATCH_REPLACE] [--patch-remove PATCH_REMOVE]
                     [-no-color] [--log-dir LOG_DIR] [-revoke-session] [-scim] [-auth] [--data DATA] [--output-access-token]
config-cli.py: error: argument --info: invalid choice: 'DefaultAuthenticationMethod' (choose from )

$ sudo python3 /opt/jans/jans-cli/config-cli.py --info DefaultAuthenticationMethod
Operation ID: get-acrs
  Description: Gets default authentication method.
Operation ID: put-acrs
  Description: Updates default authentication method.
  Schema: AuthenticationMethod

To get sample schema type /opt/jans/jans-cli/config-cli.py --schema <schma>, for example /opt/jans/jans-cli/config-cli.py --schema AuthenticationMethod

So, it seems not using sudo was the issue. Thanks @ossdhaval for the help.
